lib/haml/buffer.rb

```diff
@@ -159,7 +159,7 @@ def rstrip!
     # @since Haml 4.0.1
     # @private
     def fix_textareas!(input)
-      return input unless toplevel? && input.include?('<textarea'.freeze)
```

Comment:
Note that `toplevel?` is considered to be added just for optimization, and it must be removed now to apply this in partial view (we didn't need to call in partial before because outside layout will render partial as ruby script and call `fix_textareas!` for it). 

And this fix won't cause regression as this method is idempotent and safe to call multiple times.